src/Nonlinear/operators.jl

```diff
@@ -568,22 +624,109 @@ function eval_univariate_gradient(
     return ret::T
 end
 
+"""
+    eval_univariate_function_and_gradient(
+        registry::OperatorRegistry,
+        op::Union{Symbol,Integer},
+        x::T,
+    )::Tuple{T,T} where {T}
+
+Evaluate the function and first-derivative of the operator `op(x)::T`, where
+`op` is a univariate function in `registry`.
+
+If `op isa Integer`, then `op` is the index in
+`registry.univariate_operators[op]`.
+
+## Example
+
+```jldoctest
+julia> import MathOptInterface as MOI
+
+julia> r = MOI.Nonlinear.OperatorRegistry();
+
+julia> MOI.Nonlinear.eval_univariate_function_and_gradient(r, :abs, -1.2)
+(1.2, -1.0)
+
+julia> r.univariate_operators[3]
+:abs
+
+julia> MOI.Nonlinear.eval_univariate_function_and_gradient(r, 3, -1.2)
+(1.2, -1.0)
+```
+"""
+function eval_univariate_function_and_gradient(
+    registry::OperatorRegistry,
+    op::Symbol,
+    x::T,
+) where {T}
+    id = registry.univariate_operator_to_id[op]
+    return eval_univariate_function_and_gradient(registry, id, x)
+end
+
+function eval_univariate_function_and_gradient(
+    registry::OperatorRegistry,
+    id::Integer,
+    x::T,
+) where {T}
+    if id <= registry.univariate_user_operator_start
+        return _eval_univariate(id, x)::Tuple{T,T}
+    end
+    offset = id - registry.univariate_user_operator_start
+    operator = registry.registered_univariate_operators[offset]
+    ret_f = operator.f(x)
```

Comment:
`operator.f` and `operator.f'` both trigger a runtime method lookup. I'm wondering if adding a function barrier on `operator` would bring a speed up if there is a lot of user-defined functions